bytecomp/matching.ml

```diff
@@ -1343,13 +1343,33 @@ let matcher_constr cstr = match cstr.cstr_arity with
     | Tpat_any -> Parmatch.omegas cstr.cstr_arity @ rem
     | _        -> raise NoMatch
 
+let is_none_bs_primitve : Lambda.primitive =
+  Pccall
+    {prim_name = "#is_none";
+     prim_arity = 1;
+     prim_alloc = false;
+     prim_native_name = "";
+     prim_native_float = false}
+    
+let cstr_is_option
+    ({cstr_res = {desc ; _}} : Types.constructor_description) =
+  match desc with
+  | Tconstr(path_option,_,_) -> 
+      Path.same path_option Predef.path_option
```

Comment:
maybe not enough due to type alias?

Reply:
but it is worth invesigating, since it may give some good enough info to know it is not an option type?

Reply:
I think this is tight enough.
I thought the type information would give us the list and arity of all the other constructors. Apparently it does not. We could add it, but it seems not worth it.